File: src/types/notification.ts
import { UserSimple } from "./user";

export interface Notification {
    type: number
    title: string 
    subTitle: string 
    user: UserSimple
    read: boolean
    storyID: string
    created: string
}

export enum NotificationType {
    Comment = 1,
    Like = 2,
    Mention = 3,
    Publish = 4,
    Follow = 5,
    Reply = 6
}

File: server/pkg/models/notification.go
package models

import "time"

const (
	NotificationComment = 1
	NotificationLike    = 2
	NotificationMention = 3
	NotificationPublish = 4
	NotificationFollow  = 5
	NotificationReply   = 6
)

type Notification struct {
	Type     int         `json:"type"`
	Title    string      `json:"title"`
	SubTitle string      `json:"subTitle"`
	User     *UserSimple `json:"user"`
	Read     bool        `json:"read"`
	StoryID  string      `json:"storyID"`
	Created  time.Time   `json:"created"`
}
